lib/Driver/DarwinToolChains.cpp

```diff
@@ -492,6 +494,35 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   return II;
 }
 
+
+ToolChain::InvocationInfo
+toolchains::Darwin::constructInvocation(const ArchiveJobAction &job,
+                                        const JobContext &context) const {
+   assert(context.Output.getPrimaryOutputType() == file_types::TY_Image &&
+         "Invalid linker output type.");
+
+  ArgStringList Arguments;
+
+  // Configure the toolchain.
+  const char *LibTool = "libtool";
+
+  Arguments.push_back("-static");
+
+  addPrimaryInputsOfType(Arguments, context.Inputs, context.Args,
+                         file_types::TY_Object);
+  addInputsOfType(Arguments, context.InputActions, file_types::TY_Object);
+
+  Arguments.push_back("-o");
+
+  Arguments.push_back(
+      context.Args.MakeArgString(context.Output.getPrimaryOutputFilename()));
+
+  InvocationInfo II{LibTool, Arguments};
+  II.allowsResponseFiles = true;
```

Comment:
I don't think libtool does allow response files. It *does* have a form of filelist, though, which you could borrow from the dynamic linking logic on Darwin.